src/Microsoft.AspNet.Mvc.Core/MvcOptions.cs

```diff
@@ -163,5 +166,22 @@ public int MaxModelValidationErrors
         /// </ul>
         /// </remarks>
         public IList<IMetadataDetailsProvider> ModelMetadataDetailsProviders { get; }
+
+        /// <summary>
+        /// Gets or sets programmatic configuration for the HTML helpers and <see cref="ViewContext"/>.
+        /// </summary>
+        public HtmlHelperOptions HtmlHelperOptions
+        {
+            get
+            {
+                return _htmlHelperOptions;
```

Comment:
Do we need the backing field if we are using `param: NotNull`? Wouldn't

`public HtmlHelperOptions HtmlHelperOptions { get; [param: NotNull] set; }` work?